File: app/src/main/java/com/example/nikhil/bakingapp/widget/IngredientsRemoteViewsService.java
package com.example.nikhil.bakingapp.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

/**
 * Created by NIKHIL on 24-12-2017.
 */
public class IngredientsRemoteViewsService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return new IngredientsRemoteViewsFactory(getBaseContext(), intent);
    }
}

File: app/src/main/java/com/example/nikhil/bakingapp/widget/IngredientsRemoteViewsService.kt
package com.example.nikhil.bakingapp.widget

import android.content.Intent
import android.widget.RemoteViewsService

/**
 * Created by NIKHIL on 24-12-2017.
 */
class IngredientsRemoteViewsService : RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent): RemoteViewsService.RemoteViewsFactory {
        return IngredientsRemoteViewsFactory(baseContext)
    }
}
